### ISSUE TITLE
syntax error in script

### ISSUE THREAD

**Issue by Bhagu5491** [koalaman/shellcheck#922]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): 
- [ ] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/usr/bin/ksh
your script here
while ($0) {
        delimiter=""
        if (c++ > 0) # Evaluate and then increment c
        {
            delimiter="'$delimiter'"
        }

        match($0,/ *"[^"]*" *,|[^,]*,/)  
        s=substr($0,RSTART,RLENGTH)   
        gsub(/^ *"?|"? *,$/,"",s)   
        printf (delimiter s)         
        $0=substr($0,RLENGTH+1)     
    }

```

#### Here's what shellcheck currently says:
line 175: syntax error near unexpected token `{'
line 175: `    while ($0) {'

#### Here's what I wanted or expected to see:
Error should me removed

**Comment by koalaman** [comment]
The script you have pasted is `awk`, not `ksh` as given in the shebang. ShellCheck does not support awk.
